rewrite-json/src/test/java/org/openrewrite/json/ChangeValueTest.java

```diff
@@ -84,4 +84,214 @@ void changeArrayValue() {
           )
         );
     }
+
+    @Test
+    void changeToNumberAsString() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeValue("$.apiVersion", "\"123\"")),
+          json(
+            """
+              { "apiVersion": "v1" }
+              """,
+            """
+              { "apiVersion": "123" }
+              """
+          )
+        );
+    }
+
+    @Test
+    void changeToNumberWithoutQuotes() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeValue("$.apiVersion", "123")),
+          json(
+            """
+              { "apiVersion": "v1" }
+              """,
+            """
+              { "apiVersion": 123 }
+              """
+          )
+        );
+    }
+
+    @Test
+    void changeToBooleanAsString() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeValue("$.apiVersion", "\"true\"")),
+          json(
+            """
+              { "apiVersion": "v1" }
+              """,
+            """
+              { "apiVersion": "true" }
+              """
+          )
+        );
+    }
+
+    @Test
+    void changeToBooleanWithoutQuotes() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeValue("$.apiVersion", "true")),
+          json(
+            """
+              { "apiVersion": "v1" }
+              """,
+            """
+              { "apiVersion": true }
+              """
+          )
+        );
+    }
+
+    @Test
+    void changeToStringWithDoubleQuotes() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeValue("$.apiVersion", "\"v2\"")),
+          json(
+            """
+              { "apiVersion": "v1" }
+              """,
+            """
+              { "apiVersion": "v2" }
+              """
+          )
+        );
+    }
+
+    @Test
+    void changeToStringWithDoubleQuotesAroundSingleQuotes() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeValue("$.apiVersion", "\"'v2'\"")),
+          json(
+            """
+              { "apiVersion": "v1" }
+              """,
+            """
+              { "apiVersion": "'v2'" }
+              """
+          )
+        );
+    }
+
+    @Test
+    void changeToStringSingleQuotes() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeValue("$.apiVersion", "'v2'")),
+          json(
+            """
+              { "apiVersion": "v1" }
+              """,
+            """
+              { "apiVersion": 'v2' }
+              """
+          )
+        );
+    }
+
+    @Test
+    void changeToStringSingleQuotesAroundDoubleQuotes() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeValue("$.apiVersion", "'\"v2\"'")),
+          json(
+            """
+              { "apiVersion": "v1" }
+              """,
+            """
+              { "apiVersion": '"v2"' }
```

Comment:
```json
{ "apiVersion": '"v2"' }
```

this is invalid JSON and I would have expected the parsing to fail and then this inputed value to be treated as a string:
```json
{ "apiVersion": "'\"v2\"'" }
```